src/snowflake/telemetry/_internal/opentelemetry/exporter/otlp/proto/common/metrics_encoder.py

```diff
@@ -0,0 +1,11 @@
+from opentelemetry.exporter.otlp.proto.grpc.metric_exporter import OTLPMetricExporter
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import ExportMetricsServiceRequest
+from opentelemetry.sdk.metrics.export import MetricsData
+
+
+def encode_metrics(data: MetricsData) -> ExportMetricsServiceRequest:
```

Comment:
We should return `MetricsData`.

Reply:
This method signature matches what in the latest opentelemetry-exporter-otlp-proto-common module:

```python
# opentelemetry-python/exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/metrics_encoder/__init__.py
def encode_metrics(data: MetricsData) -> ExportMetricsServiceRequest:
```

What is the advantage of MetricsData over ExportMetricsServiceRequest?

Reply:
Consider to read the comment https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L38.

TL;DR as the name of the message tells you the `ExportMetricsServiceRequest` is specific to otel protocol not when embed the data into other protocols.

Reply:
Similar to log_encoder.py, I added a method called `serialize_metrics_data` that returns serialized bytes for a MetricsData protobuf message.